extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -724,24 +723,24 @@ void TailFile::processSingleFile(core::ProcessSession& session,
 void TailFile::updateFlowFileAttributes(const std::filesystem::path& full_file_name, const TailState& state,
                                         const std::filesystem::path& fileName, const std::string& baseName,
                                         const std::string& extension,
-                                        std::shared_ptr<core::FlowFile> &flow_file) const {
-  logger_->log_info("TailFile {} for {} bytes", fileName, flow_file->getSize());
-  std::string logName = textfragmentutils::createFileName(baseName, extension, state.position_, flow_file->getSize());
-  flow_file->setAttribute(core::SpecialFlowAttribute::PATH, state.path_.string());
-  flow_file->addAttribute(core::SpecialFlowAttribute::ABSOLUTE_PATH, full_file_name.string());
-  flow_file->setAttribute(core::SpecialFlowAttribute::FILENAME, logName);
+                                        core::FlowFile& flow_file) const {
```

Comment:
Interesting way to update the Flow File attributes switching from shared pointer to pass by reference. I see you also dereferenced the pointer as you passed *flow_file into updateFlowFileAttributes. Were you noticing problems with the shared pointer pass by reference previous approach?

Reply:
No particular problems but this makes the signature more clear and removes unnecessary restrictions, so this could help future refactors.
  this function could easily work on flowfiles owned by unique_ptrs or flowfiles on the stack
  it also ensures that the flowfile is not supposed to be a nullptr and moves the responsibility to ensure it is not null to the caller side (when it deferences the shared_ptr for example)

Reply:
@james94 passing everything by shared_ptr is a half decade old legacy of the codebase that we're trying to reduce. In most cases, we can assign each object to a clear owner, with unique_ptr, which makes it easier to reason about lifetimes, and ensures that we don't accidentally create shared_ptr ownership cycles like [MINIFICPP-839](https://issues.apache.org/jira/browse/MINIFICPP-839)